src/black/strings.py

```diff
@@ -136,7 +136,10 @@ def normalize_string_prefix(s: str, remove_u_prefix: bool = False) -> str:
     match = re.match(r"^([" + STRING_PREFIX_CHARS + r"]*)(.*)$", s, re.DOTALL)
     assert match is not None, f"failed to match string {s!r}"
     orig_prefix = match.group(1)
+
     new_prefix = orig_prefix.replace("F", "f").replace("B", "b").replace("U", "u")
+    if len(new_prefix) == 2 and "r" != new_prefix[0].lower():
```

Comment:
Just double checking - There is never a chance of > 2 here right?

- Also, lets set a const for 2 to explain it's value

Feel free to pick a better name etc.
```suggestion
    TWO_CHAR_PREFIX_LEN = 2
    if len(new_prefix) == TWO_CHAR_PREFIX_LEN and "r" != new_prefix[0].lower():
```

Reply:
Yeah about that, at least with Python 3 it's max 2. Never used Python 2 tho, but trying on an online interpreter, "rb" is invalid and "f" doesn't exist. "ru" and "ub" are also invalid. So we should be ok.

I'll fight you a bit with adding the constant, if I may. Having "two char prefix len" be 2 is both a bit too obvious and not explaining enough in my opinion. Would `len(pf) > 1` or maybe a comment saying max 2 is guaranteed be better? I won't die on this hill though 😄 so feel free to just put your foot down.

Reply:
I also don't care enough to fight this. I've just always found it easier to come back to code with consts explaining a naked integer years / even weeks later. Happy with this if other maintainers are.

Reply:
Added a comment there explaining it a bit more, that is definitely appropriate and enables a more verbose explanation. Will add the const if others also agree!

Reply:
2 Seems to be the max according to the lexical analysis page on both versions.
I think a comment would be best here, as a var name won't imo explain why it's two without being overly verbose.

While not directly related, is there any reason for the yoda condition? `... new_prefix[0].lower() != "r"` reads better to me

Reply:
I guess "r is not the first prefix char" seemed more intuitive for me when writing it, but it could be "the first char is not r" just as well!

Reply:
What about some logic along the lines of "if there's an r in the prefix, then move it to the front" which will also work for future multi-character prefixes? You could also put the rest of the prefix in alphabetical order at the same time for future consistency. Might be gold-plating though.

Reply:
Yeah, reversing is essentially a shorthand in that case. I don't expect many more prefixes to be added so I just left it at that. Are there any on the horizon? If not, I'd say it's just making it too fancy, although strictly correct 😄